### DISCUSSION TITLE
Does react-navigation@4 support in react-native@0.73.2 ?

### DISCUSSION THREAD

**Discussion by gkasireddy202** [react-navigation/react-navigation#11947] [Q&A] [answered]
Hi,

We have been using react-navigation@4 in my project for the last 5 years.
is this react-navigation@4 support in react-native:0.73.2?
My project has a react-native version:0.68.7.
Please look at the documentation screenshot of react-navigation.

<img width="981" alt="Screenshot 2024-04-15 at 4 16 26 PM" src="https://github.com/react-navigation/react-navigation/assets/90603484/71097576-c81a-4d3c-a0c1-c52908e46782">

**Comment by raajnadar** [comment, marked as answer]
It will not work with the new React Native version there are many breaking changes, which are not handled in the package due to deprecation.

https://www.npmjs.com/package/react-navigation/v/4.4.4

**Reply by gkasireddy202** [reply]
I see this information in the react-navigation documentation. What about projects that use navigation@4?
They have written their navigation structure for navigation@4 having more number of screen.
If it is difficult to upgrade to the latest navigation at a time.
<img width="818" alt="Screenshot 2024-04-23 at 1 46 18 PM" src="https://github.com/react-navigation/react-navigation/assets/90603484/2911036f-0237-491e-8b6f-5c57ca0de565">

**Reply by raajnadar** [reply]
At some point in time, everyone needs to upgrade the version, in react-navigation v5 you have a compact package which can help you preserve the old structure yet use the new version. And incrementally migrate to the new API.

https://reactnavigation.org/docs/5.x/compatibility

In 2024, you can use the Expo router package if you are planning to rewrite it. All the react-navigation heavy lifting is done by the Expo router.

**Reply by gkasireddy202** [reply]
@raajnadar  - Thanks for your reply.

Can I use the https://reactnavigation.org/docs/5.x/compatibility  without rewriting the new navigation structure and continue wth old navigation structure?

Need suggestion?

**Reply by raajnadar** [reply]
You can try and give me feedback, if that doesn't work you need to rewrite the application.

**Reply by gkasireddy202** [reply]
I will try it and send the feedback.

**Reply by gkasireddy202** [reply]
I am getting an error with tab navigation.

import { createMaterialTopTabNavigator } from 'react-navigation-tabs';
import { createStackNavigator } from '@react-navigation/stack';
import { createCompatNavigatorFactory } from '@react-navigation/compat';
import ScreenB from './ScreenB';
import ScreenC from './ScreenC';
import { Text, TouchableOpacity } from 'react-native';
const tabs = createMaterialTopTabNavigator(
    {
        ScreenB: {
            screen: ScreenB,
            navigationOptions: {
                tabBarLabel: 'Screen B',
            },
        },
        ScreenC: {
            screen: ScreenC,
            navigationOptions: {
                tabBarLabel: 'Screen C',
            },
        },
    },

    {
        tabBarOptions: {
            activeTintColor: 'red',
            inactiveTintColor: 'green',
        },
    },
    {
        navigationOptions: {
            tabBarOnPress: ({ navigation, defaultHandler }) => {
                defaultHandler();
            }
        }
    },
    {
        lazy: false
    }
);

const stack =  createCompatNavigatorFactory(createStackNavigator)({
    tab: {
        screen: tabs,

        navigationOptions: props => ({
            title: 'MY Screens',
            headerStyle: {
                backgroundColor: 'white',
                borderBottomWidth: 0.5,

            },
            headerLeft: (
                <TouchableOpacity onPress={() => {
                    {
                        props.navigation.pop();
                        props.navigation.goBack();

                    }
                }}>
                    <Text style={{ fontSize: 20, padding: 8 }}>Back</Text>
                </TouchableOpacity>
            ),
        }),
    },


});

const appStack = stack;

export default appStack;

![Simulator Screenshot - iPhone 15 Pro Max - 2024-04-24 at 15 25 32](https://github.com/react-navigation/react-navigation/assets/90603484/04960627-ad81-484d-991a-36aa38df893a)

**Reply by gkasireddy202** [reply]
I think we need to update the tab navigation to @react-navigation/bottom-tabs right?

**Reply by raajnadar** [reply]
For your last question you can refer this blog -> https://reactnavigation.org/docs/5.x/upgrading-from-4.x

**Reply by gkasireddy202** [reply]
@raajnadar - Thanks for your effort.
I am working on this navigation5.

**Reply by gkasireddy202** [reply]
@raajnadar - There is no chance to fixes bugs for navigation4 like navigation 5?

**Reply by raajnadar** [reply]
Yes v4 is discontinued, the next v7 is under development, v5 will also go into maintenance mode
